lib/libc/sys/select.c

```diff
@@ -50,8 +43,11 @@ int select(int nfds, fd_set *restrict readfds, fd_set *restrict writefds,
         FD_ZERO(readfds);
     if (writefds != NULL)
         FD_ZERO(writefds);
-    
-    ret = kevent(kq, events, nevents, events, nevents, timeout == NULL ? NULL : &timeout_ts);
+
+    if (sigmask && sigprocmask(SIG_SETMASK, sigmask, &sigs))
+        return -1;
```

Comment:
>     If sigmask is a non-null pointer, then the pselect() function shall
>     replace the signal mask of the caller by the set of signals pointed to by
>     sigmask before examining the descriptors, and shall restore the signal
>     mask of the calling thread before returning.

the question is how do we want to interpret _before_ - I'm thinking about moving it before `kqueue`.

Reply:
Moved the `sigprocmask` call before `kqueue` - https://github.com/cahirwpz/mimiker/pull/1419/commits/2dc7f82d58a0e638af3ca13518660648daae6eab.